docs/sql-data-sources-parquet.md

```diff
@@ -252,6 +252,51 @@ REFRESH TABLE my_table;
 
 </div>
 
+## Columnar Encryption
+
+
+Since Spark 3.2, columnar encryption is supported for Parquet tables with Apache Parquet 1.12.
+
+Parquet uses the envelope encryption practice, where the file parts are encrypted with “data encryption keys” (DEKs), and the DEKs are encrypted with “master encryption keys” (MEKs). The DEKs are randomly generated by Parquet for each encrypted file/column. The MEKs are generated, stored and managed in a Key Management Service (KMS) of user’s choice. Parquet-test [package](https://repo1.maven.org/maven2/org/apache/parquet/parquet-hadoop/1.12.0/parquet-hadoop-1.12.0-tests.jar) has a mock KMS implementation that allows to run column encryption and decryption without a KMS server:
+
+<div class="codetabs">
+<div data-lang="scala"  markdown="1">
+{% highlight scala %}
```

Comment:
To be complete, could you add more examples for the other languages like `python`, `SQL`, `r`?

Reply:
Spark/PME* functionality has been mass tested with Scala, so I'm mostly comfortable with this example. Also, ORC column encryption has one [example](https://github.com/apache/spark/blob/master/docs/sql-data-sources-orc.md#columnar-encryption) too, I've followed the same approach. But if needed, I can add a Python sample, this is known to work with PME too. In the future, as the community gathers experience with PME in e.g. Java or R applications, this doc could be augmented with additional examples.
(* PME - "parquet modular encryption")

Reply:
I've removed the tabs, and stressed that this an illustration-only sample that can be run without a KMS server - just `spark-shell` is sufficient. This should make a scala example a sort of an intuitive choice in this place.

Reply:
ORC column encryption has SQL example because it can work in all environments (Spark Shell/PySpark/SparkR/SQL Shell/STS). I'm curious if Apache Parquet doesn't work with SQL environment, for example Spark Thrift Server?

Reply:
Yep, I also wonder about this; but the bulk of my personal experience is with direct activation of this function - so I'm comfortable with adding these samples and parameters to the Spark documentation (since they are well tested etc). If other community members get to test and verify PME via SQL interface, I think it would be a good future addition to this section.